model/event/subscribers.go

```diff
@@ -153,10 +157,11 @@ type GithubPullRequestSubscriber struct {
 	Repo     string `bson:"repo"`
 	PRNumber int    `bson:"pr_number"`
 	Ref      string `bson:"ref"`
+	ChildId  string `bson:"child"`
 }
 
 func (s *GithubPullRequestSubscriber) String() string {
-	return fmt.Sprintf("%s-%s-%d-%s", s.Owner, s.Repo, s.PRNumber, s.Ref)
+	return fmt.Sprintf("%s-%s-%d-%s-%s", s.Owner, s.Repo, s.PRNumber, s.Ref, s.ChildId)
```

Comment:
I believe the duplicate key came from the key for the child patch and the main evergreen patch colliding. I'm assuming that if I did this multiple times, it would still collide, because the child patch Id should be unique, so it can only create one notification per child.